src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/CategoryData.php

```diff
@@ -57,9 +57,10 @@ public function addData($storeId, array $indexData)
             $categoryDataRow = array_merge(
                 $categoryDataRow,
                 [
-                    'category_id' => (int) $categoryDataRow['category_id'],
-                    'is_parent'   => (bool) $categoryDataRow['is_parent'],
-                    'name'        => (string) $categoryDataRow['name'],
+                    'category_id'   => (int) $categoryDataRow['category_id'],
+                    'category_uid'  => base64_encode((string) $categoryDataRow['category_id']),
```

Comment:
can you add some local caching ? 

and also do it with ``$this->uidEncoder->encode((string) $category->getId());`` ? This will require injecting the uidEncoder but will ensure that if Magento implementation of the encoder changes, we also benefit the change.

Reply:
Why is caching the `category_uid` necessary if every `$categoryDataRow` is unique?

Reply:
let's imagine we have product A, B and C belonging to category Y.

we will compute base64_encode(Y) three times (for each row), but we could compute it only one time and get it from cache on next calls.

Reply:
@romainruaud Added local caching.